### ISSUE TITLE
question on https://nshipster.com/xcconfig/

### ISSUE THREAD

**Issue by heckj** [NSHipster/articles#618]
I was reading through https://nshipster.com/xcconfig/ and I can see that some XCconfig files are applied per target or configuration value, but it's not clear *how* they're linked. 

The article says:
> "Once you’ve created an xcconfig file, you can assign it to one or more build configurations for its associated targets." 

but when I went and made one, it wasn't clear how or where to bind the configuration to a target, the project, etc. In some examples, I've seen the configs listed with the target names and configuration - is there a naming convention that needs to be used and maintained that makes the linkage, or is there another "target selection" mechanism that should be used?

**Comment by mattt** [comment]
Good question, @heckj. To help make this clearer, I updated the screenshots in the article to provide some more context to the UI in Xcode responsible for this.

![xcconfig-project-configurations--light-a82440e27f27e3b139ab51c7317780c6b4a017dfcc56532972da5d521f1f0988](https://user-images.githubusercontent.com/7659/58173026-10f0fc00-7c4f-11e9-9bf9-689a6aebfb0f.png)
 
As for what to call these configurations and how they correspond to targets — that's going to depend entirely on the specifics of each project, contingent on both client and server. "Development / Staging / Production" is a common naming convention; other patterns I've seen is "Development / TestFlight / App Store" and "Alpha / Beta / (and so on)".

**Comment by heckj** [comment]
Thanks a the broader screenshot with additional context definitely helps there!

**Comment by yet-another-developer** [comment]
How would one organise xcconfig if a project has multiple targets of different platform?
For example the project has targets for iOS, watchOS, and tvOS.

**Comment by hashem-intelligentb** [comment]
I copied this in to `development.xcconfig` file. but the app name didn't change
I have linked configuration file to the scheme according to the screenshot above. but still app name doesnt change.
if I replace `PRODUCT_NAME` to `APP_NAME` and provide this for `bundle name` in `info.plist` it works and app name changes!
<img width="454" alt="Screen Shot 2019-06-23 at 1 53 07 PM" src="https://user-images.githubusercontent.com/50131190/59974524-4ecea080-95be-11e9-9b9a-a0d61f68096f.png">

``` 
\\Development.xcconfig
PRODUCT_NAME = $(inherited) α
```

note updated screenshot still is not reflected in the main article

**Comment by hashem-intelligentb** [comment]
Kindly attach sample project to the article. I'm sure many have my issue

**Comment by tonyli508** [comment]
Hi @hashem-intelligentb 

> Kindly attach sample project to the article. I'm sure many have my issue

The PRODUCT_NAME didn't reflect .xcconfig files is because you already set a custom value in your `Building Settings`. Delete that custom value will fix everything.

![Screenshot 2019-07-05 at 11 02 30](https://user-images.githubusercontent.com/5927738/60715462-f7f98d00-9f14-11e9-9cc7-3f4edbb1084f.png)

**Comment by HHuckebein** [comment]
I found that (as a general rule) check, once you have you config files ready, all the build settings.
If you see entries in bold delete them as they are overriding settings specified in your config files.
----
I came across a problem where we have a dependent framework project within the main project where we needed to have a framework build as debug or release build dependent on the configuration on the configuration of the main project (Debug A, Debug B or Release A, Release B).
I created configurations with the same name in the framework project but it seems that Xcode 10.3 is always using the value stored in the buildConfiguration variable (last entry in the configuration section named "Use ... for command-line builds".
Any ideas how to solve this? Thanks
